skbeam/core/accumulators/binned_statistic.py

```diff
@@ -70,8 +70,13 @@ def __init__(self, sample, statistic='mean',
             A sequence of lower and upper bin edges to be used if the
             edges are not given explicitely in `bins`. Defaults to the
             minimum and maximum values along each dimension.
-
+        mask : array_like
+            array of ones and zeros with the same shape as `sample`.
+            Values with mask==0 will be ignored.
         """
+        if mask is None:
+            mask = np.ones_like(sample)
```

Comment:
yeah I agree with you and thought about it later. the masking probably makes most sense at the lower level, since it could be generalized to anything (and binnedstatistic expects the incoming data to be of matching shape as bins, so masking then follows naturally there).

I want to also test your mask with my code later (it might help me better understand the code), when I have access to my machine, maybe I might have some advice to add (but can't now). Yeah, we're all busy multitasking :-).